File: 3228-maximum-number-of-operations-to-move-ones-to-the-end/3228-maximum-number-of-operations-to-move-ones-to-the-end.py
class Solution:
    def maxOperations(self, s: str) -> int:
        count1 = 0
        res = 0
        i = 0
        while i < len(s):
            if s[i] == '0':
                while i + 1 < len(s) and s[i + 1] == '0':
                    i += 1
                res += count1
            else:
                count1 += 1
            i += 1
        return res

File: 3228-maximum-number-of-operations-to-move-ones-to-the-end/3228-maximum-number-of-operations-to-move-ones-to-the-end.cpp
class Solution {
public:
    int maxOperations(string s) {
        int count1 = 0;
        int res = 0;
        for (int i = 0; i < s.length(); i++) {
            if (s[i] == '0') {
                while ((i + 1) < s.length() && s[i + 1] == '0') {
                    i++;
                }
                res += count1;
            }
            else {
                count1++;
            }
        }
        return res;
    }
};